src/app/nation/nation.component.html

```diff
@@ -0,0 +1,53 @@
+
+  <h1>Add Nation </h1>
+  <form class="form-horizontal"(ngSubmit)="onSubmit(nation.value)" #nation="ngForm">
+    <div class="form-group">
+      <label for="admin_name">Admin Name</label>
+      <input type="text" style="width:50%;" class="form-control" id="admin_name" required minlength="5" maxlength="30" [(ngModel)]="nation.admin_name" name="admin_name" #admin_name="ngModel">
+      <div *ngIf="admin_name.errors && (admin_name.dirty || admin_name.touched)" class="alert alert-danger" style="width:50%;">
+        <div [hidden]="!admin_name.errors.required">
```

Comment:
It probably can be changed better way. @LKhadka had something that adds error message later based on validation

Reply:
This needs to be changed into a reactive form then you can use the validators [here](https://github.com/ole-vi/planet/blob/master/src/app/validators/custom-validators.ts). @Pr0chin can see an example [here](https://github.com/ole-vi/planet/blob/master/src/app/courses/courses.component.ts)

Reply:
@LKhadka :+1: